### ISSUE TITLE
Issues compiling on Windows with MingGW

### ISSUE THREAD

**Issue by joachimdalen** [emilk/loguru#41]
I get the following errors trying to compile on Windows using MinGW version: w64 5.0

```loguru.hpp: In function 'loguru::Text loguru::vtextprintf(const char*, va_list)': loguru.hpp:1603:46: error: 'vasprintf' was not declared in this scope int result = vasprintf(&buff, format, vlist);``` 

And

```loguru.hpp: In function 'bool loguru::create_directories(const char*)': loguru.hpp:1907:29: error: too many arguments to function 'int mkdir(const char*)' if (mkdir(file_path, 0755) == -1)```

Any suggestions?

**Comment by Subtixx** [comment]
I guess it's already fixed?

vasprintf issue in code is now:
```cpp
#ifdef _MSC_VER
		int bytes_needed = _vscprintf(format, vlist);
		CHECK_F(bytes_needed >= 0, "Bad string format: '%s'", format);
		char* buff = (char*)malloc(bytes_needed+1);
		vsnprintf(buff, bytes_needed+1, format, vlist);
		return Text(buff);
#else
		char* buff = nullptr;
		int result = vasprintf(&buff, format, vlist);
		CHECK_F(result >= 0, "Bad string format: '%s'", format);
		return Text(buff);
#endif
```

mkdir issue in code is now:
```cpp
#ifdef _MSC_VER
		if (_mkdir(file_path) == -1) {
#else
		if (mkdir(file_path, 0755) == -1) {
#endif
```

**Comment by emilk** [comment]
Fixed in #54
